lib/matplotlib/cbook/__init__.py

```diff
@@ -365,6 +389,13 @@ def process(self, s, *args, **kwargs):
                     proxy(*args, **kwargs)
                 except ReferenceError:
                     self._remove_proxy(proxy)
+                # this does not capture KeyboardInterrupt, SystemExit,
+                # and GeneratorExit
+                except Exception as e:
```

Comment:
I guess the technically correct way to do this is to store weakref.refs rather than weakref.proxies, so that we can distinguish between failure in dereferencing the callback weakref.ref and failures raised from within the callback.